authentik/events/models.py

```diff
@@ -160,6 +160,51 @@ def get_events_per(
         return self.get_queryset().get_events_per(time_since, extract, data_points)
 
 
+class EventBatch(models.Model):
+    """Model to store information about batches of events."""
+
+    batch_id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    event_type = models.CharField(max_length=255)
+    start_time = models.DateTimeField(auto_now_add=True)
+    end_time = models.DateTimeField(null=True, blank=True)
+    event_count = models.IntegerField(default=0)
+    last_updated = models.DateTimeField(auto_now=True)
+    max_batch_size = models.IntegerField(default=10)
+    batch_timeout = models.IntegerField(default=60)  # Timeout in seconds
```

Comment:
We probably need a reference to the transport too to send the notifications when the task sends it out